src/vscrollpanel.cpp

```diff
@@ -81,11 +81,24 @@ void VScrollPanel::draw(NVGcontext *ctx) {
     mChildPreferredHeight = child->preferredSize(ctx).y();
     float scrollh = height() *
         std::min(1.0f, height() / (float) mChildPreferredHeight);
+        
+    float offset = -mScroll*(mChildPreferredHeight - mSize.y());
 
     nvgSave(ctx);
     nvgTranslate(ctx, mPos.x(), mPos.y());
     nvgIntersectScissor(ctx, 0, 0, mSize.x(), mSize.y());
-    nvgTranslate(ctx, 0, -mScroll*(mChildPreferredHeight - mSize.y()));
+    nvgTranslate(ctx, 0, offset);
+
+
+    Widget *parentWindow = (Widget*)window();
```

Comment:
I'm not a big fan of the loop over child widgets. What about redesigning this patch so that your change is done in the popup button class, specifically line 66? This is the place where you could auto-detect whether one of the parents is a ScrollPanel and if so, adjust the position.